state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1200,36 +1196,49 @@ impl<T: Config<I>, I: 'static> EgressApi<T::TargetChain> for Pallet<T, I> {
 			*id
 		});
 		let egress_id = (<T as Config<I>>::TargetChain::get(), egress_counter);
-		match maybe_ccm_with_gas_budget {
+		let egress_fee = match maybe_ccm_with_gas_budget {
 			Some((
 				CcmDepositMetadata { source_chain, source_address, channel_metadata },
 				gas_budget,
-			)) => ScheduledEgressCcm::<T, I>::append(CrossChainMessage {
-				egress_id,
-				asset,
-				amount,
-				destination_address: destination_address.clone(),
-				message: channel_metadata.message,
-				cf_parameters: channel_metadata.cf_parameters,
-				source_chain,
-				source_address,
-				gas_budget,
-			}),
-			None => ScheduledEgressFetchOrTransfer::<T, I>::append(FetchOrTransfer::<
-				T::TargetChain,
-			>::Transfer {
-				asset,
-				destination_address: destination_address.clone(),
-				amount,
-				egress_id,
-			}),
-		}
+			)) => {
+				ScheduledEgressCcm::<T, I>::append(CrossChainMessage {
+					egress_id,
+					asset,
+					amount,
+					destination_address: destination_address.clone(),
+					message: channel_metadata.message,
+					cf_parameters: channel_metadata.cf_parameters,
+					source_chain,
+					source_address,
+					gas_budget,
+				});
+				gas_budget
+			},
+			None => {
+				let (amount_after_fees, egress_fee) =
+					Self::withhold_transaction_fee(IngressOrEgress::Egress, asset, amount);
+				ScheduledEgressFetchOrTransfer::<T, I>::append({
+					FetchOrTransfer::<T::TargetChain>::Transfer {
+						asset,
+						destination_address: destination_address.clone(),
+						amount: amount_after_fees,
+						egress_id,
+					}
+				});
+				egress_fee
+			},
+		};
+
+		DepositBalances::<T, I>::mutate(asset, |tracker| {
+			tracker.register_transfer(amount);
```

Comment:
Both `tracker.register_transfer` and `tracker.mark_as_fetched` used to be invoked in `fn do_egress_scheduled_fetch_transfer`. 

One of them is moved into `fn schedule_egress` as the original (gross?) amount is not saved after taking the fee.

Can the `tracker.mark_as_fetched` invocation be moved to the `fn schedule_egress` too?
IMO, it would make easier to reason about storage changes this way.